cogs/challenges.py

```diff
@@ -13,14 +13,14 @@
     STAFF_ROLE_ID,
 )
 
-
 def setup(bot):
     bot.add_cog(ChallengeHandler(bot))
 
 
 class ChallengeHandler(commands.Cog):
     def __init__(self, bot):
         self.bot = bot
+        self.code_length = {}
```

Comment:
You should store this somewhere where it isn't deleted if the bot is restarted, for example the database. Your approach could only work if when you started the bot you would fetch the submissions from the submission channel from the last message that wasn't a bot, and store the code lengths from there, then when a new submission is made you would extend the code lengths (like you do it now).